File: user-service-java/src/main/java/com/project/userService/services/UserService.java
package com.project.userService.services;

import com.project.userService.models.Role;
import com.project.userService.models.User;

import java.util.List;

public interface UserService {
    User getUserById(long id);

    List<User> getAllUsers();

    void updateUserRole(long userId, Role role);

    void saveUser(User user);
}

File: telegram-bot/services/user_service/user_service.py
from model.student import Student
from model.user import User


class UserService:
    def get_user_by_id(self, user_id: int) -> User:
        raise NotImplementedError

    def get_student_by_user_id(self, user_id: int) -> Student:
        raise NotImplementedError

    def is_student(self, user_id: int) -> bool:
        raise NotImplementedError
